requirements.txt

```diff
@@ -1,6 +1,16 @@
 ollama==0.4.4
-openai==1.56.2
-python-dotenv==1.0.1
+openai>=1.56.2
+python-dotenv>=1.0.0
 tweepy==4.14.0
 sqlalchemy==2.0.28
-click==8.1.7
+click>=8.1.7
+ratelimit>=2.2.1
+requests>=2.31.0
+typing-extensions>=4.8.0
+pytest>=7.4.0
+pytest-cov>=4.0.0
+
+# Additional testing dependencies
+pytest-check>=2.1.4
+pytest-mock>=3.12.0
+coverage>=7.3.2
```

Comment:
I am okay with additional requirements, but I prefer to fix the version; it reduces bugs.